api/ttn/lorawan/v3/networkserver_mac_settings_profile.proto

```diff
@@ -0,0 +1,137 @@
+import "thethings/flags/annotations.proto";
+import "ttn/lorawan/v3/end_device.proto";
+import "ttn/lorawan/v3/identifiers.proto";
+import "validate/validate.proto";
+
+option go_package = "go.thethings.network/lorawan-stack/v3/pkg/ttnpb";
+
+message CreateMACSettingsProfileRequest {
+  // The MAC settings profile to create.
+  MACSettingsProfile mac_settings_profile = 1 [(validate.rules).message.required = true];
+}
+
+message CreateMACSettingsProfileResponse {
+  // The MAC settings profile.
+  MACSettingsProfile mac_settings_profile = 1 [(validate.rules).message.required = true];
+}
+
+message GetMACSettingsProfileRequest {
+  // The identifiers of the MAC settings profile.
+  MACSettingsProfileIdentifiers mac_settings_profile_ids = 1 [(validate.rules).message.required = true];
+  // The names of the fields that should be returned.
+  google.protobuf.FieldMask field_mask = 2;
+}
+
+message GetMACSettingsProfileResponse {
+  // The MAC settings profile.
+  MACSettingsProfile mac_settings_profile = 1 [(validate.rules).message.required = true];
+}
+
+message UpdateMACSettingsProfileRequest {
+  // The identifiers of the MAC settings profile.
+  MACSettingsProfileIdentifiers mac_settings_profile_ids = 1 [(validate.rules).message.required = true];
+  // The MAC settings profile to update.
+  MACSettingsProfile mac_settings_profile = 2 [(validate.rules).message.required = true];
+  // The names of the fields that should be updated.
+  google.protobuf.FieldMask field_mask = 3;
+}
+
+message UpdateMACSettingsProfileResponse {
+  // The MAC settings profile.
+  MACSettingsProfile mac_settings_profile = 1 [(validate.rules).message.required = true];
+}
+
+message DeleteMACSettingsProfileRequest {
+  // The identifiers of the MAC settings profile.
+  MACSettingsProfileIdentifiers mac_settings_profile_ids = 1 [(validate.rules).message.required = true];
+}
+
+message DeleteMACSettingsProfileResponse {}
+
+message ListMACSettingsProfilesRequest {
+  option (thethings.flags.message) = {
+    select: false,
+    set: true
+  };
+  // The identifiers of the application.
+  ApplicationIdentifiers application_ids = 1 [(validate.rules).message.required = true];
+  // The names of the fields that should be returned.
+  google.protobuf.FieldMask field_mask = 2;
+  // Order the results by this field path (must be present in the field mask).
+  // Default ordering is by ID. Prepend with a minus (-) to reverse the order.
+  string order = 3 [(validate.rules).string = {
+    in: [
+      "",
+      "ids.profile_id",
+      "-ids.profile_id"
+    ]
+  }];
+  // Limit the number of results per page.
+  uint32 limit = 4 [(validate.rules).uint32.lte = 1000];
+  // Page number for pagination. 0 is interpreted as 1.
+  uint32 page = 5;
+}
+
+message ListMACSettingsProfilesResponse {
+  // The MAC settings profiles.
+  repeated MACSettingsProfile mac_settings_profiles = 1;
+  // The total number of MAC settings profiles that match the request.
+  uint32 total_count = 2;
```

Comment:
Please check that this is the convention. I think we often / always return the total count in a `x-total-count` gRPC header.

Reply:
I've checked it and indeed we are using the `x-total-count` in the header so I've removed the `total_count` from the message.